src/Workspaces/CoreTestUtilities/AsynchronousOperationListenerExtensions.cs

```diff
@@ -5,15 +5,14 @@
 #nullable enable
 
 using System.Threading.Tasks;
-using System.Windows.Threading;
 using Microsoft.CodeAnalysis.Shared.TestHooks;
 
 namespace Roslyn.Test.Utilities
 {
     public static class AsynchronousOperationListenerExtensions
     {
         internal static Task WaitAllDispatcherOperationAndTasksAsync(this IAsynchronousOperationListenerProvider provider, params string[] featureNames)
-            => ((AsynchronousOperationListenerProvider)provider).WaitAllAsync(featureNames, eventProcessingAction: () => Dispatcher.CurrentDispatcher.DoEvents());
+            => ((AsynchronousOperationListenerProvider)provider).WaitAllAsync(featureNames);
```

Comment:
The belief is the DoEvents isn't necessary because we should be yielding the thread no matter what given this is Task?

Reply:
The `DoEvents` is not necessary because we're going through JTF which tracks UI thread work requests internally.

Reply:
📝 I looked into simplifying this further, but I had less confidence in correctness as it fanned out. This seems like a reasonable intermediate perf win.